eng/common/TestResources/New-TestResources.ps1

```diff
@@ -52,12 +33,30 @@ param (
     [string] $Location = '',
 
     [Parameter()]
-    [ValidateSet('AzureCloud', 'AzureUSGovernment', 'AzureChinaCloud')]
     [string] $Environment = 'AzureCloud',
 
     [Parameter()]
     [hashtable] $AdditionalParameters,
 
+    [Parameter(ParameterSetName = 'SubscriptionConfiguration', Mandatory = $true)]
+    [string] $SubscriptionConfiguration = '',
+
+    [Parameter(ParameterSetName = 'SubscriptionConfiguration', Mandatory = $true)]
+    [ValidateNotNullOrEmpty()]
+    [string] $KeyVaultName,
```

Comment:
Why are you making a bunch of Key Vault-specific changes. I don't understand what the problem was with the original code. Please clarify *why* this needs to be changed. Those variables were already in `$PSBoundParameters`, since they are bound parameters.

Reply:
@danieljurek based on where we left the conversation I didn't expect to see anything new beyond the SubscriptionConfiguration parameter that would contain a json blob that we would parse in the script. WE shouldn't need to pass in and access the keyvault.

Reply:
Why can't you just pass the parameters? What is this `SubscriptionConfiguration`? And if you ust need to pass those to ARM templates or prepopulate script parameters (but again, why not just script parameters), there are easier and less intrusive ways of doing that. If you give me the requirements / use case, I'd be happy to show you.

Reply:
We need to keep these easily callable by customers. Only passing in a blob of content is not common in PowerShell and makes the scripts harder to use.